pure-wasm/README.md

```diff
@@ -1,5 +1,36 @@
-You'll need to generate the flat buffer defs if you need them. You need flatc version 23.1.21 or below to generate valid Rust.
+# Yggdrasil Engine (Pure WASM)
 
-Generate them with this:
+This project contains bindings for Yggdrasil that are suitable for compilation into a pure, standalone WebAssembly module. It’s designed for high portability and **does not require WASI**.
 
-`flatc --rust -o pure-wasm/src flat-buffer-defs/enabled-message.fbs`
\ No newline at end of file
+See the [Yggdrasil Java Engine](../java-engine) for an example integration.
+
+## Development
+
+### Working with Data Exchange Code
+
+> [!IMPORTANT]
+> Requires [flatc version 23.1.21](https://github.com/google/flatbuffers/releases/tag/v23.1.21)
+
+We use [FlatBuffers](https://flatbuffers.dev/) for efficient data exchange over the WASM boundary. If you're modifying the data models, you’ll need to regenerate the FlatBuffer bindings:
+
+``` bash
+flatc --rust -o pure-wasm/src flat-buffer-defs/enabled-message.fbs
+```
+
+### Building
+
+This project uses low-level bit-hacking optimizations that rely on a 32-bit architecture, so the only supported target is wasm32-unknown-unknown. Build configuration is already defined in the root project's config.toml, so you can compile the WASM module with:
+
+``` bash
+cargo build --release --target wasm32-unknown-unknown
```

Comment:
Not sure if this is or not an important detail. Rust compiler still gives you this command if it fails due to the missing target
```suggestion
rustup target add wasm32-unknown-unknown # only required once
cargo build --release --target wasm32-unknown-unknown
```

Reply:
Sure we can add this. But lemme do it earlier